test/agent.test.ts

```diff
@@ -0,0 +1,416 @@
+}
+
+async function deployMockFallback(deployer: Deployer): Promise<Contract> {
+  const artifact = await deployer.loadArtifact('MockFallback');
+  return await deployer.deploy(artifact, []);
+}
+
+async function deployMockCallback(deployer: Deployer): Promise<Contract> {
+  const artifact = await deployer.loadArtifact('MockCallback');
+  return await deployer.deploy(artifact, []);
+}
+
+describe('Agent', function () {
+  const inputsEmpty = [];
+  const logicsEmpty = [];
+  const dataEmpty = [];
+  const tokensReturnEmpty = [];
+
+  let provider;
+  let wallet;
+  let deployer;
+  let agent;
+  let router;
+  let mockERC20;
+  let mockTo;
+  let mockCallback;
+  let mockWrappedNative;
+
+  async function setUpTest() {
+    provider = Provider.getDefaultProvider();
+    wallet = new Wallet(RICH_WALLET_PK_0, provider);
+    deployer = new Deployer(hre, wallet);
+    mockWrappedNative = await deployMockWrappedNative(deployer);
+    router = await deployRouter(deployer, mockWrappedNative.address);
+    agent = await deployAgent(wallet, router);
+    mockERC20 = await deployMockERC20(deployer);
+    mockTo = await deployMockFallback(deployer);
+    mockCallback = await deployMockCallback(deployer);
+  }
+
+  before(async function () {});
+
+  beforeEach(async function () {});
+
+  afterEach(async function () {});
+
+  describe('Normal', function () {
+    beforeEach(async function () {
+      await setUpTest();
+    });
+
+    it.skip('router', async function () {
+      // const deployer = RICH_WALLET_ADDR_0;
+      // expect(await agent.router()).to.be.eq(router.address);
+    });
+
+    it.skip('wrapped native', async function () {
+      // expect(await agent.wrappedNative()).to.be.eq(mockWrappedNative.address);
+    });
+
+    it('wrap before fixed amounts', async function () {
+      const amount1 = 1;
+      const amount2 = 1;
+      const amount = amount1 + amount2;
+      const input0 = {
+        token: mockWrappedNative.address,
+        balanceBps: BPS_NOT_USED,
+        amountOrOffset: amount1,
+      };
+      const input1 = {
+        token: mockWrappedNative.address,
+        balanceBps: BPS_NOT_USED,
+        amountOrOffset: amount2,
+      };
+      const inputs = [input0, input1];
+
+      const logic0 = {
+        to: mockTo.address,
+        data: abiCoder.encode(['bytes'], ['0x']),
+        inputs: inputs,
+        wrapMode: WRAP_BEFORE,
+        approveTo: ZERO_ADDRESS,
+        callback: ZERO_ADDRESS,
+      };
+
+      const user = wallet.address;
+      const logics = [logic0];
+      const receipt = await router.execute(logics, tokensReturnEmpty, SIGNER_REFERRAL, {
+        from: user,
+        value: amount,
+      });
+
+      await receipt.wait();
+      // const agent = await router.getAgent(user);
+      expect((await mockWrappedNative.balanceOf(agent)).toString()).to.be.eq(amount.toString());
+    });
+
+    it('wrap before with token', async function () {
```

Comment:
Messing up `testWrapBeforeReplacedAmounts` and `testWrapBeforeWithToken`?

Reply:
fixed